djstripe/models/core.py

```diff
@@ -1218,6 +1224,331 @@ def is_valid_object(cls, data):
+		)
+	)
+	statement_descriptor = models.CharField(
+		max_length=255,
+		null=True,
+		blank=True,
+		help_text=(
+			"Extra information about a PaymentIntent. This will appear on your customer’s "
+			"statement when this PaymentIntent succeeds in creating a charge."
+		)
+	)
+	status = StripeEnumField(
+		enum=enums.PaymentIntentStatus,
+		help_text=(
+			"Status of this PaymentIntent, one of requires_payment_method, requires_confirmation, "
+			"requires_action, processing, requires_capture, canceled, or succeeded. "
+			"You can read more about PaymentIntent statuses here."
+		)
+	)
+	transfer_data = JSONField(
+		null=True,
+		blank=True,
+		help_text=(
+			"The data with which to automatically create a Transfer when the payment is finalized. "
+			"See the PaymentIntents Connect usage guide for details."
+		)
+	)
+	transfer_group = models.CharField(
+		max_length=255,
+		help_text=(
+			"A string that identifies the resulting payment as part of a group. See the "
+			"PaymentIntents Connect usage guide for details."
+		)
+	)
+
+	def update(self, api_key=None, **kwargs):
+		"""
+		Call the stripe API's modify operation for this model
+
+		:param api_key: The api key to use for this request. Defaults to djstripe_settings.STRIPE_SECRET_KEY.
+		:type api_key: string
+		"""
+		api_key = api_key or self.default_api_key
+
+		return self.api_retrieve(api_key=api_key).modify(**kwargs)
+
+	def _api_cancel(self, api_key=None, **kwargs):
+		"""
+		Call the stripe API's cancel operation for this model
+
+		:param api_key: The api key to use for this request. Defaults to djstripe_settings.STRIPE_SECRET_KEY.
+		:type api_key: string
+		"""
+		api_key = api_key or self.default_api_key
+
+		return self.api_retrieve(api_key=api_key).cancel(**kwargs)
+
+	def _api_confirm(self, api_key=None, **kwargs):
+		"""
+		Call the stripe API's confirm operation for this model.
+
+		Confirm that your customer intends to pay with current or
+		provided payment method. Upon confirmation, the PaymentIntent
+		will attempt to initiate a payment.
+
+		:param api_key: The api key to use for this request. Defaults to djstripe_settings.STRIPE_SECRET_KEY.
+		:type api_key: string
+		"""
+		api_key = api_key or self.default_api_key
+
+		return self.api_retrieve(api_key=api_key).confirm(**kwargs)
+
+
+class SetupIntent(StripeModel):
+	"""
+	A SetupIntent guides you through the process of setting up a customer's payment credentials
+	for future payments. For example, you could use a SetupIntent to set up your customer's
+	card without immediately collecting a payment. Later, you can use PaymentIntents
+	to drive the payment flow.
+
+	NOTE: You should not maintain long-lived, unconfirmed SetupIntents.
+	For security purposes, SetupIntents older than 24 hours may no longer be valid.
+
+	Stripe documentation: https://stripe.com/docs/api#setup_intents
+	"""
+
+	stripe_class = stripe.SetupIntent
+	stripe_dashboard_item_name = "setup intents"
+
+	application = models.CharField(
```

Comment:
Could this be a ForeignKey + new model?

Reply:
I am confused on this one about the mention of `Connect Application` [ in the API here ](https://stripe.com/docs/api/setup_intents/object#setup_intent_object-application) as I couldn't find mention of `Application` in the API elsewhere. Help on this appreciated :)

Reply:
Maybe leave it un-implemented for now with a TODO, if we're not sure?

Reply:
Fixed in 2965902713

Reply:
Impressive work!! Congrats!! re. this setup intent field, it has the same meaning as `payment_intent.application` [[api-ref]](https://stripe.com/docs/api/payment_intents/object#payment_intent_object-application): the value is the unique identifier assigned to the _Connect_ application. It's, for instance, the `client_id` informed in `/oauth/authorize` [docs](https://stripe.com/docs/connect/oauth-reference#get-authorize)

Reply:
@lx-stripe ah, so we can treat it as an opaque string that doesn't correspond to an API object?

Reply:
yup... technically it belongs to the account but, since there's no way to retrieve the application client id over the API, an opaque string would be the right choice imho

Reply:
OK @tiwariayush we don't need to TODO for this one then